bundles/persistence/org.openhab.persistence.jdbc/java/org/openhab/persistence/jdbc/db/JdbcBaseDAO.java

```diff
@@ -0,0 +1,530 @@
+	}
+	
+	public void doDeleteItemsEntry(ItemsVO vo) {
+		String sql = StringUtilsExt.replaceArrayMerge(
+				SQL_DELETE_ITEMS_ENTRY, 
+				new String[]{ "#itemname#"}, new String[]{vo.getItemname()}
+				);
+		logger.debug("JDBC::doDeleteItemsEntry sql={}", sql);
+		Yank.execute(sql, null);
+	}
+	
+	public List<ItemsVO> doGetItemIDTableNames(ItemsVO vo) {
+		String sql = StringUtilsExt.replaceArrayMerge(
+				SQL_GET_ITEMID_TABLE_NAMES, 
+				new String[]{ "#itemsManageTable#"}, 
+				new String[]{ vo.getItemsManageTable()}
+				);
+		logger.debug("JDBC::doGetItemIDTableNames sql={}", sql);
+	    return Yank.queryBeanList(sql, ItemsVO.class, null);
+	}
+	
+	public List<ItemsVO> doGetItemTables(ItemsVO vo) {
+		String sql = StringUtilsExt.replaceArrayMerge(
+				SQL_GET_ITEM_TABLES, 
+				new String[]{ "#jdbcUriDatabaseName#", "#itemsManageTable#"}, 
+				new String[]{ vo.getJdbcUriDatabaseName(), vo.getItemsManageTable()}
+				);
+		logger.debug("JDBC::doGetItemTables sql={}", sql);
+	    return Yank.queryBeanList(sql, ItemsVO.class, null);
+	}
+	
+	/*************
+	 * ITEM DAOs *
+	 *************/
+	public void doUpdateItemTableNames(List<ItemVO> vol) {
+		String sql = updateItemTableNamesProvider(vol);
+		Yank.execute(sql, null);
+	}
+	
+	public void doCreateItemTable(ItemVO vo) {
+		String sql = StringUtilsExt.replaceArrayMerge(
+				SQL_CREATE_ITEM_TABLE, 
+				new String[]{ "#tableName#", "#dbType#"}, 
+				new String[]{ vo.getTableName(), vo.getDbType()}
+				);
+		Yank.execute(sql, null);
+	}
+	
+	public void doStoreItemValue(Item item, ItemVO vo) {
+		vo = storeItemValueProvider(item, vo);
+		String sql = StringUtilsExt.replaceArrayMerge(
+				SQL_INSERT_ITEM_VALUE, 
+				new String[]{ "#tableName#"}, 
+				new String[]{ vo.getTableName()}
+				);
+		Object[] params = new Object[] { vo.getValue(), vo.getValue() };
+		logger.debug("JDBC::doStoreItemValue sql={} value='{}'", sql, vo.getValue() );
+		Yank.execute(sql, params);
+	}
+	
+	public List<HistoricItem> doGetHistItemFilterQuery(Item item, FilterCriteria filter, int numberDecimalcount, String table, String name) {
+		String sql = histItemFilterQueryProvider(filter, numberDecimalcount, table, name);
+		logger.debug("JDBC::doGetHistItemFilterQuery sql={}", sql);
+		List<Object[]> m = Yank.queryObjectArrays(sql, null);
+
+		List<HistoricItem> items = new ArrayList<HistoricItem>();
+		for(int i=0; i<m.size(); i++){
+			items.add(new JdbcItem(item.getName(), getState(item, m.get(i)[1]), objectAsDate( m.get(i)[0])));
+		}
+	    return items;
+	}
+	
+	/*************
+	 * Providers *
+	 *************/
+	private String histItemFilterQueryProvider(FilterCriteria filter, int numberDecimalcount, String table, String simpleName) {
+		logger.debug("JDBC::getHistItemFilterQueryProvider filter = {}, numberDecimalcount = {}, table = {}, simpleName = {}", filter.toString(), numberDecimalcount, table, simpleName);
+		
+		SimpleDateFormat jdbcDateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```

Comment:
That class isn't threadsafe, so making it static could lead to mysterious issues.